share/completions/go.fish

```diff
@@ -7,16 +7,25 @@ complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -s a -d
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -s n -d 'print the commands but do not run them'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -s p -r -d 'number parallel builds (default=#cpus)'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o race -d 'enable data race detection'
+complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o msan -d 'enable interoperation with memory sanitizer'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -s v -d 'print packages being built'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o work -d 'print and preserve work directory'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -s x -d 'print the commands'
-complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o ccflags -r -d 'c compiler flags'
+complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o asmflags -d 'arguments to pass on each go tool asm invocation'
+complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o buildmode -x -d 'build mode to use' -a '{archive,c-archive,c-shared,default,shared,exe,pie,plugin}'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o compiler -x -d 'name of compiler to use, as in runtime' -a 'gccgo gc'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o gccgoflags -r -d 'gccgo compiler/linker flags'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o gcflags -r -d 'go compiler flags'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o installsuffix -r -d 'suffix for installation directory'
 complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o ldflags -r -d 'linker flags'
+complete -c go -n "__fish_seen_subcommand_from $__go_cmds_w_buildflags" -o linkshared -r -d 'build code that will be linked against shared libraries previously'
```

Comment:
I'm afraid I don't understand what this means. Are the shared libraries built first, and then the code is linked against them?

Reply:
Emm, that is `go help build`'s origin manual.
That's mean build a shared library, does not contain existing shared libraries code.

Reply:
The full help is

```
build code that will be linked against shared libraries previously created with -buildmode=shared.
```

I'd shorten it to `Link against shared libraries previously created with -buildmode=shared`

Reply:
Okay, you stopped in a bit of a weird spot. The full original text is:

>build code that will be linked against shared libraries previously created with -buildmode=shared.

So the point is that the code you build will use shared libraries you built before.

So something like "use pre-built shared libraries"? Maybe "use pre-built shared libraries (-buildmode=shared)"?